File: DataStructure/leetcode1546/solveLeetcode1546.go
// func maxNonOverlapping(nums []int, target int) int {
// 	prefix := 0
// 	hash := make(map[int]int)
// 	hash[0] = 1

// 	ans := 0
// 	t := 0
// 	for _, x := range nums {
// 		prefix += x

// 		t = prefix - target
// 		if _, ok := hash[t]; ok {
// 			hash = make(map[int]int)
// 			hash[0] = 1
// 			ans++
// 			prefix = 0
// 		} else {
// 			hash[prefix]++
// 		}
// 	}

// 	return ans
// }

File: DataStructure/prefix/leetcode1546/solveLeetcode1546.cpp
// class Solution 
// {
// public:
//     int maxNonOverlapping(vector<int>& nums, int target) 
//     {
//         int prefix=0;
//         unordered_map<int,int>hash{{0,1}};


//         int ans=0,t;
//         for(const auto&x:nums)
//         {
//             prefix+=x;

//             t=prefix-target;
//             if(hash.count(t))
//             {
//                 hash=unordered_map<int,int>{{0,1}};
//                 ++ans;
//                 prefix=0;
//             }
//             else ++hash[prefix];

//             //printf("%-3d",prefix);
//         }
//         //printf("\n");
//         return ans;
//     }
// };